commands.go

```diff
@@ -11,6 +11,19 @@ import (
 	yaml "gopkg.in/yaml.v2"
 )
 
+func mergedTaskSetOpts(ctx, cmdLine CmdLine) TaskSetOpt {
```

Comment:
I like how much code this saves, but there is a potential footgun we could prevent. Before you return the `TaskSetOpt`, can we make sure that the `cmdLine.Cmd` field is one of the expected commands where this combination of options is valid? I would not want another contributor to assume we can pass this in every case. Perhaps a for loop checking the command against the named constants.

Reply:
I think there should be one place per thing. It's not up to a callee to validate the caller is within a certain set of expectations. That sets a weird precedent.
Instead callers should simply not call functions that they shouldn't call. Such a mistake will easily be picked up in review.

Reply:
I suppose you're right. I dont' feel strongly, so I'll dismiss this review. Otherwise lgtm  :+1: